src/screens/ChangeWalletModal.js

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback } from 'react';
+import { View } from 'react-native';
+import { useNavigation } from 'react-navigation-hooks';
+import { useDispatch } from 'react-redux';
+import WalletList from '../components/change-wallet/WalletList';
+import { Modal } from '../components/modal';
+import WalletTypes from '../helpers/walletTypes';
+import { useAccountSettings, useInitializeWallet, useWallets } from '../hooks';
+import {
+  addressSetSelected,
+  createAccountForWallet,
+  walletsSetSelected,
+  walletsUpdate,
+} from '../redux/wallets';
+import { logger } from '../utils';
+
+const walletRowHeight = 54;
+
+const ChangeWalletModal = () => {
+  const { wallets, selected: selectedWallet } = useWallets();
+
+  const { goBack, navigate } = useNavigation();
+  const dispatch = useDispatch();
+  const { accountAddress } = useAccountSettings();
+  const initializeWallet = useInitializeWallet();
+  let rowsCount = 0;
+  if (wallets) {
+    Object.keys(wallets).forEach(key => {
+      // Wallet header
+      // If there's more than account we group them by wallet
+      if (wallets[key].addresses.length > 1) {
+        rowsCount += 1;
+      }
+      // Addresses
+      rowsCount += wallets[key].addresses.filter(account => account.visible)
+        .length;
+
+      // Add account
+      if (
+        [WalletTypes.mnemonic, WalletTypes.seed].indexOf(wallets[key].type) !==
+        -1
+      ) {
+        rowsCount += 1;
+      }
+    });
+    // Import wallet
+    rowsCount += 1;
+  }
+  let listHeight = walletRowHeight * rowsCount;
+  if (listHeight > 298) {
+    listHeight = 298;
+  }
+
+  const onChangeAccount = useCallback(
+    async (wallet_id, address) => {
+      try {
+        const wallet = wallets[wallet_id];
+        dispatch(walletsSetSelected(wallet));
+        dispatch(addressSetSelected(address));
+        await initializeWallet();
+        goBack();
+      } catch (e) {
+        logger.log('error while switching account', e);
+      }
+    },
+    [dispatch, goBack, initializeWallet, wallets]
+  );
+
+  const onEditWallet = useCallback(
+    id => {
+      const wallet = wallets[id];
+      if (!selectedWallet) return;
+
+      let isDeletable = false;
+      if (Object.keys(wallet).length > 1 && selectedWallet.id !== id) {
+        isDeletable = true;
+      }
+      navigate('ExpandedAssetScreen', {
+        actionType: 'Import',
+        address: undefined,
+        asset: [],
+        isDeletable,
+        onCloseModal: async args => {
```

Comment:
onCloseModal is called by the cancel function in WalletProfileCreator without any args, which still triggers the onSuccess to toggle the importing state

Reply:
This is related to my response here: https://github.com/rainbow-me/rainbow/pull/606#issuecomment-624941336